wgpu-hal/src/vulkan/mod.rs

```diff
@@ -149,6 +140,103 @@ pub struct Instance {
     shared: Arc<InstanceShared>,
 }
 
+/// The semaphores used to synchronize the swapchain image acquisition.
+///
+/// One of these is created per swapchain image and is used for the swapchain
```

Comment:
This is unclear. A swapchain has several images. Each has its own `SwapchainSemaphores`. What does it mean for all of them to be "used for the swapchain"?